File: old_trunk/cost/iona/orbacus/java/poa/Test_impl.java
// **********************************************************************
//
// Copyright (c) 2000
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;

final class Test_impl extends TestPOA
{
    //
    // From TestBase (no multiple inheritance)
    //
    public static void
    TEST(boolean expr)
    {
	if(!expr)
	    throw new test.common.TestException();
    }

    private POA poa_;
    org.omg.PortableServer.Current current_;
    private String name_;
    private boolean compare_;

    Test_impl(ORB orb, String name, boolean compare)
    {
	name_ = name;
	compare_ = compare;

	org.omg.CORBA.Object currentObj = null;

	try
	{
	    currentObj = orb.resolve_initial_references("POACurrent");
	}
	catch(org.omg.CORBA.ORBPackage.InvalidName ex)
	{
	}

	TEST(currentObj != null);
	current_ =  org.omg.PortableServer.CurrentHelper.narrow(currentObj);
	TEST(current_ != null);
    }

    Test_impl(ORB orb, POA poa)
    {
	poa_ = poa;
	name_ = "";
	compare_ = false;

	org.omg.CORBA.Object currentObj = null;

	try
	{
	    currentObj = orb.resolve_initial_references("POACurrent");
	}
	catch(org.omg.CORBA.ORBPackage.InvalidName ex)
	{
	}

	TEST(currentObj != null);
	current_ =  org.omg.PortableServer.CurrentHelper.narrow(currentObj);
	TEST(current_ != null);
    }

    public void
    aMethod()
    {
	if(compare_)
	{
	    byte[] oid = null;
	    try
	    {
		oid = current_.get_object_id();
	    }
	    catch(org.omg.PortableServer.CurrentPackage.NoContext ex)
	    {
		throw new RuntimeException();
	    }

	    String oidString = new String(oid);

	    TEST(oidString.equals(name_));
	}
    }

    public POA
    _default_POA()
    {
	if(poa_ != null)
	    return poa_;
	return super._default_POA();
    }
}

File: cost/iona/orbacus/cpp/poa/Test_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <Test_skel.h>

#include <Test_impl.h>

using namespace CORBA;
using namespace PortableServer;

Test_impl::Test_impl(ORB_ptr orb, const char* name, Boolean compare)
    : name_(name), compare_(compare)
{
    Object_var currentObj;

    try
    {
        currentObj = orb -> resolve_initial_references("POACurrent");
    }
    catch(const ORB::InvalidName&)
    {
    }

    TEST(!is_nil(currentObj));
    current_ = PortableServer::Current::_narrow(currentObj);
    TEST(!is_nil(current_));
}

Test_impl::Test_impl(ORB_ptr orb, POA_ptr poa)
    : poa_(POA::_duplicate(poa)), name_((const char*)""), compare_(false)
{
    Object_var currentObj;

    try
    {
        currentObj = orb -> resolve_initial_references("POACurrent");
    }
    catch(const ORB::InvalidName&)
    {
    }

    TEST(!is_nil(currentObj));
    current_ = PortableServer::Current::_narrow(currentObj);
    TEST(!is_nil(current_));
}

void
Test_impl::aMethod()
    throw(SystemException)
{
    if(compare_)
    {
	ObjectId_var oid;
	try
	{
	    oid = current_ -> get_object_id();
	}
	catch(const PortableServer::Current::NoContext&)
	{
	}

        String_var oidString = ObjectId_to_string(oid);
        
        TEST(strcmp(oidString, name_) == 0);
    }
}

POA_ptr
Test_impl::_default_POA()
{
    if(!is_nil(poa_))
        return POA::_duplicate(poa_);
    return ServantBase::_default_POA();
}
